spec/mongoid/relations/embedded/many_spec.rb

```diff
@@ -2748,26 +2748,47 @@ class TrackingIdValidationHistory
       end
 
       let!(:location) do
-        address.locations.create(name: "work")
+        address.locations.create(name: "vacation", number: 0)
+        address.locations.create(name: "work", number: 3)
       end
 
-      context "when updating with a hash" do
+      context "when updating with replacement of embedded array" do
 
-        before do
-          address.update_attributes(locations: [{ name: "home" }])
+        context "when updating with a hash" do
+
+          before do
+            address.update_attributes(locations: [{ name: "home" }])
+          end
+
+          it "updates the attributes" do
+            expect(address.locations.first.name).to eq("home")
+          end
+
+          it "overwrites the existing documents" do
+            expect(address.locations.count).to eq(1)
+          end
+
+          it "persists the changes" do
+            expect(address.reload.locations.count).to eq(1)
+          end
         end
+      end
+
+      context "when updating a field in a document of the embedded array" do
 
-        it "updates the attributes" do
-          expect(address.locations.first.name).to eq("home")
+        before do
+          location.number = 7
+          location.save
         end
 
-        it "overwrites the existing documents" do
-          expect(address.locations.count).to eq(1)
+        it "persists the change" do
+          expect(person.reload.addresses.first.locations.find(location.id).number).to eq(7)
```

Comment:
I would put this in a let block so you would have something like:

```ruby
let(:updated_attribute) do
  person.reload.addresses.first.locations.find(location.id).number
end
```

and then 
```ruby
expect(updated_attribute).to eq(7)
```